core/monitor/monitor.go

```diff
@@ -19,54 +19,60 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/google/keytransparency/core/client/mutationclient"
+	"github.com/google/keytransparency/core/client"
 	"github.com/google/keytransparency/core/monitorstorage"
 
 	"github.com/google/trillian"
-	"github.com/google/trillian/client"
 	"github.com/google/trillian/types"
 
 	"github.com/golang/glog"
 
 	pb "github.com/google/keytransparency/core/api/v1/keytransparency_proto"
+	tclient "github.com/google/trillian/client"
 	tcrypto "github.com/google/trillian/crypto"
 )
 
 // Monitor holds the internal state for a monitor accessing the mutations API
 // and for verifying its responses.
 type Monitor struct {
-	mClient     pb.KeyTransparencyClient
+	cli         *client.Client
 	signer      *tcrypto.Signer
 	trusted     types.LogRootV1
-	logVerifier *client.LogVerifier
-	mapVerifier *client.MapVerifier
+	logVerifier *tclient.LogVerifier
+	mapVerifier *tclient.MapVerifier
 	store       monitorstorage.Interface
 }
 
 // NewFromDomain produces a new monitor from a Domain object.
-func NewFromDomain(mClient pb.KeyTransparencyClient,
+func NewFromDomain(cli pb.KeyTransparencyClient,
 	config *pb.Domain,
 	signer *tcrypto.Signer,
 	store monitorstorage.Interface) (*Monitor, error) {
-	logVerifier, err := client.NewLogVerifierFromTree(config.GetLog())
+	logVerifier, err := tclient.NewLogVerifierFromTree(config.GetLog())
 	if err != nil {
 		return nil, fmt.Errorf("could not initialize log verifier: %v", err)
 	}
-	mapVerifier, err := client.NewMapVerifierFromTree(config.GetMap())
+	mapVerifier, err := tclient.NewMapVerifierFromTree(config.GetMap())
 	if err != nil {
 		return nil, fmt.Errorf("could not initialize map verifier: %v", err)
 	}
-	return New(mClient, logVerifier, mapVerifier, signer, store)
+
+	ktClient, err := client.NewFromConfig(cli, config)
+	if err != nil {
+		return nil, fmt.Errorf("could not create kt client: %v", err)
+	}
+
+	return New(ktClient, logVerifier, mapVerifier, signer, store)
 }
 
 // New creates a new instance of the monitor.
-func New(mClient pb.KeyTransparencyClient,
-	logVerifier *client.LogVerifier,
-	mapVerifier *client.MapVerifier,
+func New(cli *client.Client,
+	logVerifier *tclient.LogVerifier,
+	mapVerifier *tclient.MapVerifier,
 	signer *tcrypto.Signer,
```

Comment:
nit: Make param order and struct field order consistent?

Reply:
done